scrapyrt/utils.py

```diff
@@ -0,0 +1,14 @@
+import inspect
+from scrapy import Request
+
+
+def extract_scrapy_request_args(dictionary, raise_error=False):
+    result = dictionary.copy()
+    args = inspect.getargspec(Request.__init__).args
+    for key in dictionary.keys():
+        if key not in args:
+            result.pop(key)
+            if raise_error:
+                msg = u"{!r} is not a valid argument for scrapy.Request.__init__"
+                raise ValueError(msg.format(key))
+    return result
```

Comment:
... unless `result.pop()` was put in original version intentionally to split passed dictionary into spider data and request data. Please document this and let's keep version with `.pop()`